### PR TITLE
Okta mfa

### PR DESCRIPTION
**Pull request by VarsityShortcut**
tested

### CODE REVIEW

File: foxpass-radius-agent.py [hunk 1]
```diff
@@ -121,9 +143,76 @@ def two_factor(username):
...
+    resp_json = okta_request(url, headers, timeout, post=True)
+
+    # poll for transaction completion
+    while True:
+        if resp_json['factorResult'] == 'SUCCESS':
+            return True
+        elif resp_json['factorResult'] == 'WAITING':
+            url = resp_json['_links']['poll']['href']
+            # sleep for 1 second to rate limit requests
+            time.sleep(1)
```
Comment (aren) [inline foxpass-radius-agent.py]: Please read http://www.gevent.org/gevent.monkey.html and make sure time.sleep() is covered by monkey.patch_all() above.

File: foxpass-radius-agent.py [hunk 2]
```diff
@@ -121,9 +143,76 @@ def two_factor(username):
     if response and response['result'] == 'allow':
         return True
 
-    logger.info("Duo failed")
+    logger.info("Duo mfa failed")
+    return False
+
+def okta_mfa(username):
+    # if Duo is not configured, return success
```
Comment (aren) [inline foxpass-radius-agent.py]: Okta

File: foxpass-radius-agent.py [hunk 3]
```diff
@@ -121,9 +143,76 @@ def two_factor(username):
...
+    if not get_config_item('okta_hostname') or \
+       not get_config_item('okta_apikey'):
+        logger.info("Okta MFA not configured")
+        return True
+
+    hostname = get_config_item('okta_hostname')
+    api_key = get_config_item('okta_apikey')
+
+    # optional timeout on requests, not related to overall transaction
+    timeout = get_config_item('okta_request_timeout')
```
Comment (aren) [inline foxpass-radius-agent.py]: get_config_item's second parameter is the default

File: foxpass-radius-agent.py (hunk 1 above)
Comment (VarsityShortcut) [reply foxpass-radius-agent.py]: it looks like it is covered but i'm not really sure what it does. maybe explain face to face

**Pull request merged**